python/cudf/cudf/core/column/column.py

```diff
@@ -77,37 +78,27 @@ def data_array_view(self):
 
         if is_categorical_dtype(self.dtype):
             return self.codes.data_array_view
-        else:
-            dtype = self.dtype
 
-        result = rmm.device_array_from_ptr(
-            ptr=self.data.ptr, nelem=len(self), dtype=dtype
-        )
-        result.gpu_data._obj = self
+        result = cp.asarray(self.data)
         return result
 
     @property
     def mask_array_view(self):
         """
         View the mask as a device array
         """
-        result = rmm.device_array_from_ptr(
-            ptr=self.mask.ptr,
-            nelem=calc_chunk_size(len(self), mask_bitsize),
-            dtype=np.int8,
-        )
-        result.gpu_data._obj = self
+        result = cp.asarray(self.mask)
         return result
 
     def __len__(self):
         return self.size
 
     def to_pandas(self):
         arr = self.data_array_view
-        sr = pd.Series(arr.copy_to_host())
+        sr = pd.Series(cp.asnumpy(cp.asarray(arr)))
```

Comment:
Any reason for the double `cp.asarray(...)` here? We know `arr` will always be a cupy array.

Reply:
Probably none. Was just trying to get the tests to pass. Will drop it.

I think soonish we will want to just replace this with the RMM work we've been doing.

Reply:
Fixed